core/trino-main/src/main/java/io/trino/sql/routine/SqlRoutinePlanner.java

```diff
@@ -187,27 +187,26 @@ protected IrStatement visitIfStatement(IfStatement node, Context context)
             IrStatement statement = null;
 
             List<ElseIfClause> elseIfList = node.getElseIfClauses().reversed();
+            Optional<IrStatement> elseStatement = node.getElseClause().map(clause -> block(statements(clause.getStatements(), context)));
             for (int i = 0; i < elseIfList.size(); i++) {
                 ElseIfClause elseIf = elseIfList.get(i);
                 RowExpression condition = toRowExpression(context, elseIf.getExpression());
                 IrStatement ifTrue = block(statements(elseIf.getStatements(), context));
 
                 Optional<IrStatement> ifFalse = Optional.empty();
-                if ((i == 0) && node.getElseClause().isPresent()) {
-                    List<ControlStatement> elseList = node.getElseClause().get().getStatements();
-                    ifFalse = Optional.of(block(statements(elseList, context)));
+                if ((i == 0) && elseStatement.isPresent()) {
+                    ifFalse = elseStatement;
                 }
                 else if (statement != null) {
                     ifFalse = Optional.of(statement);
                 }
```

Comment:
It'd be cleaner to do this, instead:

```java
Optional<IrStatement> ifFalse;
if (statement == null) {
    ifFalse = elseStatement;
}
else {
    ifFalse = Optional.of(statement);
}
```

`statement` can only be null in the first loop, so there's no reason to check for `i == 0`.

Reply:
You could also do:

```java
Optional<IrStatement> ifFalse = Optional.ofNullable(statement).or(() -> elseStatement);                        ;
```

but I think that is a bit more cryptic.

Reply:
An even simpler implementation (for the whole method):

```java
Optional<IrStatement> ifFalse = node.getElseClause().map(clause -> block(statements(clause.getStatements(), context)));

for (ElseIfClause elseIf : node.getElseIfClauses().reversed()) {
    ifFalse = Optional.of(new IrIf(
            toRowExpression(context, elseIf.getExpression()),
            block(statements(elseIf.getStatements(), context)),
            ifFalse));
}

return new IrIf(
        toRowExpression(context, node.getExpression()),
        block(statements(node.getStatements(), context)),
        ifFalse);
```

Reply:
That is much better, updated.